common.c

```diff
@@ -105,3 +114,78 @@ const Instr_t Factorial[PROGRAM_SIZE] = {
     Instr_Print,    // n
     Instr_Halt
 };
+
+cpu_t init_cpu () {
+    cpu_t cpu = {.pc = 0, .sp = -1, .state = Cpu_Running,
+                 .steps = 0, .stack = {0},
+                 .pmem = LoadedProgram ? LoadedProgram : DefProgram};
+    return cpu;
+}
+
+static const char *steplimit_opt = "--steplimit=";
+static const char *inp_prog_opt = "--inp-prog=";
+
+static inline
+void report_usage_and_exit(char * exec_name, int ret_code) {
+    fprintf(stderr, "Usage: %s %s<num> %s<str>\n", exec_name, steplimit_opt, inp_prog_opt);
+    exit (ret_code);
+}
+
+long long parse_args(int argc, char** argv) {
+    long long steplimit = LLONG_MAX;
+    FILE *prog_file = NULL;
+
+    for (int i = 1; i < argc; ++i) {
+        if (!strcmp(argv[i], "--help"))
+            report_usage_and_exit(argv[0], 2);
+        else if (!strncmp(argv[i], steplimit_opt, strlen(steplimit_opt))) {
+            char *endptr = NULL;
+            steplimit = strtoll(argv[i] + strlen(steplimit_opt), &endptr, 10);
+            if (errno || (*endptr != '\0')) {
+                fprintf(stderr, "Unrecognized steplimit: %s\a", argv[i]);
+                report_usage_and_exit(argv[0], 2);
+            }
+        }
+        else if (!strncmp(argv[i], inp_prog_opt, strlen(inp_prog_opt))) {
+            prog_file = fopen(argv[i] + strlen(inp_prog_opt), "rb");
+            if (errno || prog_file == NULL) {
+                fprintf(stderr, "Unrecognized program file: %s\n", argv[i]);
+                report_usage_and_exit(argv[0], 2);
+            }
+        }
+        else {
+            fprintf(stderr, "Unrecognized option: %s\n", argv[i]);
```

Comment:
Please also add an ability to pass `steplimit` without any keyword. I.e., 

    ./switched 1000

Should work just as 

    ./switched --steplimit=1000

Such variant of steplimit can be handled after all keyword arguments (starting with "--") are processed

This is required because several BAT scripts already assume that it is possible to pass `steplimit` as positional argument, and your current commit will break them. Moreover, it is shorter to type number for steplimit without a keyword during debugging time.